components/KiboBillingAddress.vue

```diff
@@ -0,0 +1,186 @@
+        @input="updateField('countryCode', $event)"
+      >
+        <SfSelectOption
+          v-for="countryOption in countries"
+          :key="countryOption"
+          :value="countryOption"
+        >
+          {{ countryOption }}
+        </SfSelectOption>
+      </SfSelect>
+      <SfInput
+        :label="$t('Phone Number')"
+        name="phone"
+        class="form__element"
+        required
+        :value="billingDetails.phoneNumbers && billingDetails.phoneNumbers.home"
+        @input="updateField('phoneNumbers', $event)"
+      />
+      <SfCheckbox
+        v-model="saveBillingAddress"
+        :label="$t('Save Billing Address')"
+        name="getBillingAddress"
+        class="form__element form__checkbox"
+        @change="$emit('saveBillingAddress', saveBillingAddress)"
+      />
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { SfInput, SfSelect, SfCheckbox } from "@storefront-ui/vue"
+import { watch, ref } from "@nuxtjs/composition-api"
+export default {
+  name: "KiboBillingAddress",
+  components: {
+    SfInput,
+    SfSelect,
+    SfCheckbox,
+  },
+  props: {
+    value: {
+      type: Object,
+      default: () => ({}),
+    },
+    countries: {
+      type: Array,
+      default: () => [],
+    },
+    shipping: {
+      type: Object,
+      default: () => ({}),
+    },
+  },
+
+  setup(props, context) {
+    const billingDetails = ref({ ...props.value })
+    const sameAsShipping = ref(false)
+    const saveBillingAddress = ref(false)
+
+    const updateField = (fieldName, fieldValue) => {
+      const address = [
+        "address1",
+        "address2",
+        "cityOrTown",
+        "stateOrProvince",
+        "postalOrZipCode",
+        "countryCode",
+      ]
+
+      if (fieldName === "phoneNumbers") {
+        billingDetails.value.phoneNumbers = {
+          ...billingDetails.value.phoneNumbers,
+          home: fieldValue,
+        }
+      } else if (address.includes(fieldName)) {
+        billingDetails.value.address = {
+          ...billingDetails.value.address,
+          [fieldName]: fieldValue,
+        }
+      } else {
+        billingDetails.value[fieldName] = fieldValue
+      }
+
+      context.emit("billingAddressData", {
+        ...billingDetails.value,
+      })
+    }
+    watch(sameAsShipping, () => {
+      if (sameAsShipping.value) {
+        Object.keys(props.shipping).forEach((key) => {
+          billingDetails.value[key] = props.shipping[key]
+        })
+      } else {
+        Object.keys(props.shipping).forEach((key) => {
+          billingDetails.value[key] = ""
+        })
+      }
+      context.emit("billingAddressData", { ...billingDetails.value })
+    })
+
+    return {
+      updateField,
+      billingDetails,
+      sameAsShipping,
+      saveBillingAddress,
+    }
+  },
+}
+</script>
+
+<style lang="scss" scoped>
+@import "~@storefront-ui/shared/styles/components/templates/SfPayment.scss";
```

Comment:
Check if this is still needed after CSS optimization or not?

Reply:
I have verified but its require still we will check